yarpl/include/yarpl/Refcounted.h

```diff
@@ -66,9 +71,40 @@ class Reference {
   Reference() = default;
   inline /* implicit */ Reference(std::nullptr_t) {}
 
+  // TODO: Move check for count() into this constructor
+  // TODO: Remove public ctor for Reference(T*)
   explicit Reference(T* pointer) : pointer_(pointer) {
     inc();
   }
+  explicit Reference(T* pointer, detail::skip_initial_refcount_check)
```

Comment:
Are these ctors meant to be private?  If so then I approve of the `detail::` namespace.  But if they're meant to be private to `Reference` then I think we shouldn't use the namespace.